Refactor OSX.yml, now inputs can be provided

carlopi opened the pull request:
Additional inputs that can be provided:
- override_git_describe (to be used as environment OVERRIDE_GIT_DESCRIBE)
- git_ref (to be used if ones wants to build for a given hash or tag that is different than the current branch)
- skip_tests (to avoid running tests)

All three defaults to '', and this is used a bit in a hack since '' works well as default (git checkout '' is a no-op).

Advantages of the changes:
- no real code change (good!)
- moving out of the `startsWith(github.ref, 'refs/tags/v')` checks that will only trigger on release day into something that can be triggered regularly ${{ inputs.skip_tests != 'true' }}. To note that the only value that will actually skip tests is 'true', all other will have the test run
- allowing to externally provide (via a not yet PRd workflow) the OVERRIDE_GIT_DESCRIBE variable via workflow_dispatch OR workflow_call

Feedback would be great.

If this makes sense I would be for rolling this out the other workflows that generate release binaries, then potentially also wider.

Part of the wider move to staged releases.

carlopi committed:
Refactor OSX.yml, now inputs can be provided

Mytherin reviewed (commented):
Looks great - one minor comment

Mytherin commented on .github/workflows/OSX.yml:
```diff
@@ -1,7 +1,21 @@
 name: OSX
 on:
+  workflow_call:
+    inputs:
```
Can this be duplicate eliminated and/or do we need both call and dispatch?

samansmink replied:
welcome to gh actions 🥲 https://github.com/orgs/community/discussions/39357

carlopi replied:
Tricked Mark into asking this question, since this works as documentation.

We can get rid of one of the two, possibly workflow_dispatch that can be implemented using workflow_call + helper workflow, BUT I think it's also good to keep this duplication [for now?].

samansmink reviewed (commented):
Looks good, one thing: I'm not sure if the Github Actions are run from the original commit or the version that is now selected with the git_ref. It seems like it will pick the actions from the version that you check out to: https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#adding-an-action-from-the-same-repository 

Its just something to keep in mind: assuming this workflow is called at a different version than the one building it, there could theoretically be incompatibilities between the action and the called workflow. (e.g. changed parameters) I think this is not really an issue though

carlopi commented:
> Looks good, one thing: I'm not sure if the Github Actions are run from the original commit or the version that is now selected with the git_ref. It seems like it will pick the actions from the version that you check out to: https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#adding-an-action-from-the-same-repository
> 
> Its just something to keep in mind: assuming this workflow is called at a different version than the one building it, there could theoretically be incompatibilities between the action and the called workflow. (e.g. changed parameters) I think this is not really an issue though

Should adding some logging make this more easier on future us?

I added git_ref for flexibility given I was already here, but yes, it should be used with care.

carlopi committed:
Add upload-assets-to-staging.sh script and invoke it from OSX

carlopi committed:
Add workflow to invoke other (working name InvokeCI.yml)

carlopi committed:
[Dev] Fix 3f41dcba07c: detection of GIT_COMMIT_HASH

carlopi commented:
Solved the comments, and bundled with this also the upload-assets-to-staging script.

My idea is for rolling out the staging of artifacts is using OSX as test-bed.
Once this is merged, artifacts will be saved on every run of OSX workflow on main, that means:
1 set of artifacts every night will be uploaded.

In addition, via the InvokeCI.yml workflow (naming things is hard, suggestions are welcome), OSX.yml (and other workflows to follow) can be invoked passing the intended target (say `v0.10.1` as parameter). That would generate another set of artifacts.

Artifacts will be saved in the artifacts s3 bucket, using as folders: `duckdb/v0.10.1/.` in one case (overridden by newer runs) or `duckdb/v0.10.0-1225-GIT_HASH/.` in the other case.

I think at first always uploading is safer / better to double check what's going on.

Mytherin commented:
Thanks!
